### ISSUE TITLE
JSR 303 - @valid annotation is not getting generated on the List of complex type

### ISSUE THREAD

**Issue by gr423** [mulesoft-labs/raml-for-jax-rs#361]
Hi Team ,

I've the below RAML

```
event:
type: object
properties:
  metadataset:
   required: false
   type: metadatSetType[]

metadatSetType:
  type: object
  properties:
    locale:
    required: true
    type: string
  name:
    required:false
    type: string
```

Generated Java Objects  are as below ,

I'm expecting the @valid annotation on the metadataSet of event .Which is not getting generated.
And the JsonMapperConfiguration is never considered while generating the code .
includeAdditionalproperties is made false , but still the java objects generated has the field.

```
@JsonInclude(JsonInclude.Include.NON_NULL)
Class Event
{
    @JsonProperty("metadataSet")
    List<MetadatSetType> metadataSet;
  @JsonIgnore
  private Map<String, Object> additionalProperties = new HashMap<String, Object>();
}

```
```
@JsonInclude(JsonInclude.Include.NON_NULL)
Class MetadatasetType
{
 @JsonProperty("locale")
 @NotNull
  String locale;
 @JsonProperty("name")
String name;
  @JsonIgnore
  private Map<String, Object> additionalProperties = new HashMap<String, Object>();
}
```
Pom File 
```

<plugin>
<groupId>org.raml.jaxrs</groupId>
<artifactId>raml-to-jaxrs-maven-plugin</artifactId>
<version>3.0.1</version>
<executions>
<execution>
<id>GenerateAdminV2</id>
<goals>
<goal>generate</goal>
</goals>
<phase>generate-sources</phase>
<configuration>
<ramlFile>${basedir}/src/main/resources/raml/v1/MAS_api.raml</ramlFile>
<resourcePackage>com..mas.v1.resource</resourcePackage>
<modelPackage>com..mas.v1.model</modelPackage>
<removeOldOutput>true</removeOldOutput>
<generateTypesWith>
	<value>jsr303</value>
	<value>jackson2</value>
   </generateTypesWith>
  <jsonMapperConfiguration>
	 <includeAdditionalProperties>false</includeAdditionalProperties>
</jsonMapperConfiguration>
</configuration>
</execution>
</executions>
</plugin>

```

**Comment by gr423** [comment]
@jstoiko 
Hi Jonathan,
Please can you validate the issue .
Basically i want to validate each element of the collection , which is provided by the Javax validation .
But since the @valid annotation is missed on the List type in the generated code, i'm  missing on the validation of the List elements .


Thanks in advance,
Ramya

**Comment by jpbelang** [comment]
> includeAdditionalproperties is made false , but still the java objects generated has the field.

Have you tried the current snapshot ?  I've fixed that, I think.

https://github.com/mulesoft-labs/raml-java-tools/issues/12

**Comment by jpbelang** [comment]
For the @Valid annotation, I will add it if the list contains an object type or union type (as opposed to a list type or primitive type).  

String and integers require custom annotations.  I know that java 8 supports annotating generic types, but right now, we have to support java 7 and this would be a major change.  You could still pull it off with a plugin.

**Comment by jpbelang** [comment]
@gr423 , could you try the SNAPSHOT please ?  (On branch 3.0.3)

**Comment by gr423** [comment]
@jpbelang , 
Its working  :) Thank you !!
The @Valid annotation is getting generated on the list type and the validations are working.
I did take the  the release 3.0.3 snapshot version and built the project.
Since i build the raml-to-jaxrs project locally i was able to use raml-to-jaxrs-maven-plugin with 
3.0.3-SNAPSHOT version .

Can we expect the same version in maven repository .
Sorry i'm not sure of the procedure of when the latest version of the projects gets reflected in the maven repo.Can i get some clarity on this .

Thanks and Regards,
Ramya

**Comment by jpbelang** [comment]
The SNAPSHOTS are on mulesoft's repositories (I don't have them here, I'll get them for you tonight).

Building them locally is certainly ok, but if you need to share them between devs, either you upload them to your own repo or point to the mulesoft repo

**Comment by gr423** [comment]
Sure, let me try this option.


Thank you

**Comment by gr423** [comment]
@jpbelang  
Hi  , 

Since our project would go to production by this month end ,i'm not able to proceed with the SNAPSHOT version  .

Please could you let me know , if can expect the release version some time closer.

Or else can you suggest me the alternate, with which i can go ahead doing validations  .

Thanks in advance ,
Ramya S

**Comment by jpbelang** [comment]
I have one bit of the Arrays bug that I need to finish, which I should do tomorrow.

I'm trying to release on the week-end.

**Comment by gr423** [comment]
Thanks for the update .
Sure we will wait for the release .

Reagrds,
Ramya

**Comment by jpbelang** [comment]
I've just released 3.0.3.  it should come up soon on central.

**Comment by gr423** [comment]
@jpbelang 
Hi ,
The RAML exposed endpoints are consumed by CXF Rest Client.
But getting the issue  of deserialization on retrieving the response .
The issue is asked for in stackOverflow.
https://stackoverflow.com/questions/52521191/jax-rs-client-cant-deserialize-json-response-returned-from-raml-1-0-generated-cl

Please could you give me some insight here .
```

Below is generated interface.
@Path("/editorialChannels")
public interface EditorialChannels {
  /**
   */
  @PUT
  @Produces("application/json")
  @Consumes("application/json")
  PutEditorialChannelsResponse putEditorialChannels(
      @Valid com.nagra.opentv.mas.v1.model.EditorialChannels entity);

  class PutEditorialChannelsResponse extends ResponseDelegate {
    private PutEditorialChannelsResponse(Response response, Object entity) {
      super(response, entity);
    }

    private PutEditorialChannelsResponse(Response response) {
      super(response);
    }

    public static PutEditorialChannelsResponse respond200WithApplicationJson(PutResponse entity) {
      Response.ResponseBuilder responseBuilder = Response.status(200).header("Content-Type", "application/json");
      responseBuilder.entity(entity);
      return new PutEditorialChannelsResponse(responseBuilder.build(), entity);
    }
}
}

```
